rskj-core/src/main/java/co/rsk/net/sync/PeersInformation.java

```diff
@@ -0,0 +1,80 @@
+package co.rsk.net.sync;
+
+import co.rsk.net.MessageChannel;
+import co.rsk.net.NodeID;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+
+/**
+ * This is mostly a workaround because SyncProcessor needs to access MessageChannel instances.
+ * TODO(mc) remove this after the logical node abstraction is created, since it will wrap
+ *     things such as the underlying communication channel.
+ */
+public class PeersInformation {
+    private final SyncConfiguration syncConfiguration;
+    private Map<NodeID, SyncPeerStatus> peerStatuses = new HashMap<>();
+
+    public PeersInformation(SyncConfiguration syncConfiguration){
+        this.syncConfiguration = syncConfiguration;
+    }
+
+    public int count() {
+        return peerStatuses.size();
+    }
+
+    public int countIf(Predicate<SyncPeerStatus> predicate) {
+        long count = peerStatuses.values().stream()
+                .filter(predicate)
+                .count();
+        return Math.toIntExact(count);
+    }
+
+    public SyncPeerStatus getOrRegisterPeer(MessageChannel messageChannel) {
+        SyncPeerStatus peerStatus = this.peerStatuses.get(messageChannel.getPeerNodeID());
+
+        if (peerStatus != null && !peerStatus.isExpired(syncConfiguration.getExpirationTimePeerStatus()))
+            return peerStatus;
+
+        return this.registerPeer(messageChannel);
+    }
+
+    public SyncPeerStatus getPeer(NodeID nodeID) {
+        // TODO(mc) check expiration
+        return this.peerStatuses.get(nodeID);
+    }
+
+    public Optional<MessageChannel> getBestPeer() {
+        return peerStatuses.entrySet().stream()
+                .filter(e -> !e.getValue().isExpired(syncConfiguration.getExpirationTimePeerStatus()))
+                .max(this::bestPeerComparator)
+                .map(Map.Entry::getValue)
+                .map(SyncPeerStatus::getMessageChannel);
+    }
+
+    public Set<NodeID> knownNodeIds() {
+        return peerStatuses.keySet();
+    }
+
+    private int bestPeerComparator(Map.Entry<NodeID, SyncPeerStatus> left, Map.Entry<NodeID, SyncPeerStatus> right) {
+         return Long.compare(
```

Comment:
The best peer is the one with the highest block number?

Reply:
This is improved on a later branch